src/types.rs

```diff
@@ -532,6 +586,42 @@ impl SourceMap {
         }
     }
 
+    /// Given a location, name and minified source file resolve a minified
+    /// name to an original function name.
+    ///
+    /// This invokes some guesswork and requires access to the original minified
+    /// source.  This will not yield proper results for anonymous functions or
+    /// functions that do not have clear function names.  (For instance it's
+    /// recommended that dotted function names are not passed to this
+    /// function).
+    pub fn get_original_function_name(&self, line: u32, col: u32,
+                                      minified_name: &str, source: &str) -> Option<&str> {
+        // fast way out if we are not looking up a valid javascript identifier
+        if !is_valid_javascript_identifier(minified_name) {
+            return None;
+        }
+        let mut token_opt = self.lookup_token(line, col);
+        while let Some(token) = token_opt.take() {
```

Comment:
idea here is that we scan backwards in the file until we find our identifier lead by the `function` token. So if one searches for `i` we would try to find the `i` in `function i` to have a better chance at getting the right name.

Maybe we should add a maximum token search range here so we do not walk back 20 MB of javascript code?